dask/_task_spec.py

```diff
@@ -857,7 +865,11 @@ def _check_data_node(tsk):
         deps = dependencies[key]
         dependents_key = dependents[key]
 
-        if len(deps) != 1 and len(dependents_key) != 1 or _check_data_node(dsk[key]):
+        if (
+            len(deps) != 1
+            and len(dependents_key) != 1
+            or _check_non_fusable_node(dsk[key])
```

Comment:
very, very small nit: I know this is already the old version but generally I recommend to not have a function dispatch in a hot loop like this. that's unnecessary overhead 

adding a comment offer a similar way to communicate this without paying for the function dispatch
```suggestion           
            # check non fusable
            or tsk.block_fusion
            or (isinstance(tsk, DataNode)
            and not isinstance(tsk.value, str))
```

Reply:
Yeah, I'll rip out the data thing now and then we can inline